control-plane/agents/src/bin/core/controller/reconciler/poller.rs

```diff
@@ -98,6 +98,16 @@ impl ReconcilerWorker {
             };
         }
     }
+    fn start_trigger(&self, registry: &Registry) -> PollTriggerEvent {
+        let Some(sim_args) = registry.sim_args() else {
+            return PollTriggerEvent::Start;
+        };
+        if sim_args.no_start_event {
+            PollTriggerEvent::SimulStart
+        } else {
```

Comment:
Likewise here, anything that's on the code maybe a debug_assertion or a feature?